packages/gunshi/src/plugin.ts

```
@@ -70,144 +71,166 @@
-  decorateCommand<L extends Record<string, any> = {}>(
+  decorateCommand<
+    L extends Record<string, unknown> = {},
+    C extends Record<string, unknown> = keyof G['extensions'] extends never
+      ? L
+      : G['extensions'] & L
+  >(
     decorator: (
       baseRunner: (
-        ctx: CommandContextWithExt<A, keyof E extends never ? L : E & L>
+        ctx: Readonly<CommandContext<GunshiParams<{ args: G['args']; extensions: C }>>>
       ) => Awaitable<void | string>
```

Biome (1.9.4)

[error] void is confusing inside a union type.

Unsafe fix: Use undefined instead.


(lint/suspicious/noConfusingVoidType)